### PR TITLE
add citest to PRs; get detailed test results

### PR DESCRIPTION
**Pull request by richm**
Use `ci NUM` to add the `[citest]` comment to a PR
Use `t NUM` to show the detailed checks and test results - only shows
tests/checks which were not successful.

### CODE REVIEW

File: role-merge-prs.sh [hunk 1]
```diff
@@ -39,33 +39,63 @@ list_prs() {
...
-get_check_summary() {
-    local pr total success failure pending cancelled error conclusion state name
+get_checks() {
+    local pr total success failure pending cancelled error conclusion state name status display_name
     pr="$1"
     gh pr view "$pr" -R "$origin_org/$repo" --json statusCheckRollup \
-      --template '{{range .statusCheckRollup}}{{print (or .conclusion "<nil>")}} {{print (or .state "<nil>")}} {{println .name}}{{end}}' | {
-    while read -r conclusion state name; do
+      --template '{{range .statusCheckRollup}}{{print (or .conclusion "<nil>")}}#{{print (or .state "<nil>")}}#{{print .name}}#{{println .context}}{{end}}' | \
+    while IFS="#" read -r conclusion state name context; do
```
Comment (spetrosi) [inline role-merge-prs.sh:47]: Does this assume that PRs do not have `#` in their name?

File: role-merge-prs.sh [hunk 2]
```diff
@@ -39,33 +39,63 @@ list_prs() {
...
+            display_name="$context"
+        fi
+        echo "${display_name}#${status}"
+    done
+}
+
+get_check_detail() {
+    local pr name status
+    pr="$1"; shift
+    get_checks "$pr" | while IFS=# read -r name status; do
```
Comment (spetrosi) [inline role-merge-prs.sh:78]:
```suggestion
    get_checks "$pr" | while IFS="#" read -r name status; do
```
You do this above, with quotes it's better

File: role-merge-prs.sh [hunk 3]
```diff
@@ -39,33 +39,63 @@ list_prs() {
...
+    done
+}
+
+get_check_summary() {
+    local pr
+    pr="$1"
+    get_checks "$pr" | {
+    local -A counts
+    local total status name
+    while IFS=# read -r name status; do
```
Comment (spetrosi) [inline role-merge-prs.sh:94]:
```suggestion
    while IFS="#" read -r name status; do
```

File: role-merge-prs.sh (hunk 1 above)
Comment (richm) [reply role-merge-prs.sh:47]: not PRs, but the labels/text we use for the checks/tests - e.g. 
```json
    {
      "__typename": "CheckRun",
      "completedAt": "2023-05-23T00:44:11Z",
      "conclusion": "SUCCESS",
      "detailsUrl": "https://github.com/linux-system-roles/network/actions/runs/5051822786/jobs/9064065108",
      "name": "ansible_lint",
      "startedAt": "2023-05-23T00:42:44Z",
      "status": "COMPLETED",
      "workflowName": "Ansible Lint"
    }
```
or
```json
    {
      "__typename": "StatusContext",
      "context": "CentOS-7-latest/ansible-2.14",
      "startedAt": "2023-05-23T22:46:19Z",
      "state": "FAILURE",
      "targetUrl": "https://dl.fedoraproject.org/pub/alt/linuxsystemroles/logs/lsr-citool_network-611-9c5b88b_CentOS-7-latest_20230523-224601/artifacts/summary.html"
    }
```
right now it seems pretty safe to use `#` as the delimiter - doesn't look like any `name` or `context` uses that character

**Pull request merged**